interfaces/console/src/app/console/sites/[id]/page.tsx

```diff
@@ -302,66 +318,105 @@ const Page: React.FC<SiteDetailsProps> = ({ params }) => {
+      fetchPolicy: 'network-only',
+      onCompleted: (data) => {
+        if (data.getMetricsStat.metrics.length > 0) {
+          setNodeUptimes((prev) => {
+            const newUptimes = { ...prev };
+            data.getMetricsStat.metrics.forEach((m) => {
+              if (m.type === NODE_UPTIME_KEY) {
+                newUptimes[m.nodeId] = m.value;
+              }
+            });
+            return newUptimes;
+          });
+
+          if (statVar && statVar.data && statVar.data.from) {
+            const from = statVar.data.from;
+            const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.AllNode}-${data.getMetricsStat.metrics[0].nodeId}-${from}`;
+            MetricStatSubscription({
+              key: sKey,
+              nodeId: data.getMetricsStat.metrics[0].nodeId,
+              userId: user.id,
+              url: env.METRIC_URL,
+              orgName: user.orgName,
+              type: Stats_Type.AllNode,
+              from: from,
+            });
+            PubSub.subscribe(sKey, handleStatSubscription);
           }
+        }
+      },
+      onError: (err) => {
+        console.error('Error fetching node metrics:', err);
+        setSnackbarMessage({
+          id: 'node-metrics-err-msg',
+          message: err.message,
+          type: 'error',
+          show: true,
         });
-
-        const sKey = `stat-${user.orgName}-${user.id}-${Stats_Type.AllNode}-${statVar?.data.from ?? 0}`;
-        MetricStatSubscription({
-          key: sKey,
-          nodeId: id,
-          userId: user.id,
-          url: env.METRIC_URL,
-          orgName: user.orgName,
-          type: Stats_Type.AllNode,
-          from: statVar?.data.from ?? 0,
-        });
-        PubSub.subscribe(sKey, handleStatSubscription);
-      }
-    },
-  });
+      },
+    });
 
   useEffect(() => {
     const to = getUnixTime();
     const from = to - STAT_STEP_29;
-    if (!id) {
-      setSnackbarMessage({
-        id: 'node-not-found-msg',
-        message: 'Node not found.',
-        type: 'error',
-        show: true,
-      });
-      router.back();
-    } else if (id) {
-      const to = getUnixTime();
-      const from = to - STAT_STEP_29;
-      getMetricStat({
-        variables: {
-          data: {
-            to: to,
-            nodeId: id,
-            from: from,
-            userId: user.id,
-            step: STAT_STEP_29,
-            orgName: user.orgName,
-            withSubscription: true,
-            type: Stats_Type.AllNode,
+
+    if (nodeIds.length > 0) {
+      nodeIds.forEach((nodeId) => {
+        getMetricStat({
+          variables: {
+            data: {
+              to: to,
+              nodeId: nodeId,
+              from: from,
+              userId: user.id,
+              step: STAT_STEP_29,
+              orgName: user.orgName,
+              withSubscription: true,
+              type: Stats_Type.AllNode,
+            },
           },
-        },
+        });
```

Comment:
As i mentioned in my last PR review, Rather looping through `nodes/sites` on console and making multiple metrics call for each `node/site` we should have handle it on BFF. 

Purpose of BFF is to handle all these complex logic and provide data which console need.

By doing that way we'll get multiple advantages too. No need for multiple API calls for every node/site. Node need to manage stat for each node/site as apollo `data` response is a store itself.